src/file/files.js

```diff
@@ -6,244 +6,155 @@ import Private from '../private'
   }
 
-  xhr.send(encoded ? options.data : Utils.stringToBiteArray(getBuilder(options.fileName, options.data, boundary)))
+  return content
+}
 
-  if (async) {
-    return xhr
+const getContentSize = content => {
+  if (content.size) {
+    return content.size
   }
 
-  if (xhr.status >= 200 && xhr.status < 300) {
-    return xhr.responseText ? JSON.parse(xhr.responseText) : true
-  } else {
-    throw badResponse(xhr)
-  }
+  return content.options && content.options.knownLength
 }
 
-function getBuilder(filename, filedata, boundary) {
-  const dashdash = '--'
-  const crlf = '\r\n'
-  let builder = ''
-
-  builder += dashdash
-  builder += boundary
-  builder += crlf
-  builder += 'Content-Disposition: form-data; name="file"'
-  builder += '; filename="' + filename + '"'
-  builder += crlf
-
-  builder += 'Content-Type: application/octet-stream'
-  builder += crlf
-  builder += crlf
-
-  builder += filedata
-  builder += crlf
+const sendFile = options => {
+  const url = Urls.filePath(options.path) + '/' + sanitizeFileName(options.fileName)
+  const query = {}
 
-  builder += dashdash
-  builder += boundary
-  builder += dashdash
-  builder += crlf
+  if (Utils.isBoolean(options.overwrite)) {
+    query.overwrite = options.overwrite
+  }
 
-  return builder
+  return Backendless._ajax({
+    method      : 'POST',
+    url         : url,
+    query       : query,
+    form        : { file: options.file },
+    asyncHandler: options.asyncHandler
+  })
 }
 
 export default class Files {
   constructor() {
     // this.restUrl = Urls.files()
   }
 
+  /**
+   * @param {String} path
+   * @param {String} fileName
+   * @param {String|Uint8Array} fileContent
+   * @param {Boolean} overwrite
+   * @returns {Promise.<String>}
+   */
   saveFile = Utils.promisified('_saveFile')
 
+  /**
+   * @deprecated All sync API methods are deprecated
+   * @param {String} path
+   * @param {String} fileName
+   * @param {String|Uint8Array} fileContent
+   * @param {Boolean} overwrite
+   * @returns {String}
+   */
   saveFileSync = Utils.synchronized('_saveFile')
 
-  _saveFile(path, fileName, fileContent, overwrite, async) {
+  _saveFile(path, fileName, fileContent, overwrite, asyncHandler) {
     if (!path || !Utils.isString(path)) {
       throw new Error('Missing value for the "path" argument. The argument must contain a string value')
     }
 
-    if (!fileName || !Utils.isString(path)) {
+    if (!fileName || !Utils.isString(fileName)) {
       throw new Error('Missing value for the "fileName" argument. The argument must contain a string value')
     }
 
     if (overwrite instanceof Async) {
-      async = overwrite
+      asyncHandler = overwrite
       overwrite = null
     }
 
-    if (typeof File !== 'undefined' && !(fileContent instanceof File)) {
-      fileContent = new Blob([fileContent])
-    }
+    fileContent = toBiteArray(fileContent)
 
-    if (fileContent.size > 2800000) {
+    if (getContentSize(fileContent) > 2800000) {
```

Comment:
please move this value `2800000` as constant to the top file

Reply:
done